### PR TITLE
build: change executable name into "cataclysm-bn-tiles"

### PR DESCRIPTION
**Pull request by Ali-Anomma**
### Required

- [X] I wrote the PR title in [conventional commit format](https://docs.cataclysmbn.org/en/contribute/changelog_guidelines/).
- [X] I ran the [code formatter](https://docs.cataclysmbn.org/en/contribute/contributing/#code-style).
- [X] I linked any relevant issues using [github keyword syntax](https://docs.cataclysmbn.org/en/contribute/contributing/#pull-request-notes) so it can be closed automatically.
- [X] I have [committed my changes to new branch that isn't `main`](https://docs.cataclysmbn.org/en/contribute/contributing/#make-your-changes) so it won't cause conflict when updating `main` branch later.

### Optional
- [X] This is a PR that modifies build process or code organization.
  - [X] Please document the changes in the appropriate location in the `doc/` folder.
## Purpose of change

Apps detecting games, such as Discord considers this game as Cataclysm DDA as they both has same exe name. So i added -bn to this fork, giving a way to differ.

## Describe the solution

Replaced cataclysm-tiles with cataclysm-bn-tiles and replaced cataclysm on cmake files with cataclysm-bn. 

## Describe alternatives you've considered

Look deeply on game detecting apps and make them recognize CBN by just from window name, not the executable one.

## Testing

MSYS2 tiled version showed bn name both on console and task manager. Additional OS and curses tests are welcome.

## Additional context

Should help with #3497.
Discord saying it gets game data from https://www.igdb.com/ , as Discord-recognized CDDA has its own page on it: https://www.igdb.com/games/cataclysm-dark-days-ahead .
We need to do some off site work for recognition.

### CODE REVIEW

**Review by chaosvolt — DISMISSED** [review]
1. Compiled and load-tested.
2. Discord no longer has it pop up as anything.

image

I assume that's an improvement at least over it mislabeling us as DDA.

File: cataclysm-launcher [hunk 1]
```diff
@@ -37,7 +37,7 @@ cd "$DIR" || {
 if [ ! -f "$BIN" ] || [ "$BIN" = "cataclysm-launcher" ]
 then
     BIN=
-    for bin in cataclysm-tiles cataclysm
+    for bin in cataclysm-bn-tiles cataclysm
```
Comment (scarf005) [inline cataclysm-launcher:40]:
```suggestion
    for bin in cataclysm-bn-tiles cataclysm-bn
```

**Review by scarf005 — CHANGES_REQUESTED** [review]
could you check all `cataclysm` and `cataclysm-bn` is changed accordingly?

**Comment by Ali-Anomma** [comment]
> 1. Compiled and load-tested.
> 2. Discord no longer has it pop up as anything.
> 
> image
> 
> I assume that's an improvement at least over it mislabeling us as DDA.

go to discord settings>registered games>add game
there is something named with random numbers and letters (name of CBN instance after cataclysm bright nights part). select it.
igbd registeration needed for universal "Cataclysm: Bright Nights" text for everyone

**Comment by RoyalFox2140** [comment]
> 1. Compiled and load-tested.
> 2. Discord no longer has it pop up as anything.
> 
> image
> 
> I assume that's an improvement at least over it mislabeling us as DDA.

Discord manually sets up every connection for the user on their end (Which I am still livid about, you can't change it once they do it) They will not create the connection unless someone emails them about it or gets in contact with someone directly.

**Comment by chaosvolt** [comment]
Nice, that seems fine to me then.

**Review by chaosvolt — APPROVED** [review]

**Review by scarf005 — APPROVED** [review]
let's see how it turn out...

**Pull request merged**